src/prefect/variables.py

```diff
@@ -0,0 +1,72 @@
+import sys
+from types import ModuleType
+from typing import Optional
+
+from prefect._internal.concurrency.api import create_call, from_sync
+from prefect.client.orchestration import get_client
+
+
+def get(name: str, default: str = None) -> Optional[str]:
+    """
+    Get a variable by name from a sync context.
+
+    ```
+        from prefect import variables
+
+        @flow
+        def my_flow():
+            var = variables.get("my_var")
+    ```
+    """
+    variable = get_variable_by_name(name)
+    return variable.value if variable else default
+
+
+async def aget(name: str, default: str = None) -> Optional[str]:
+    """
+    Get a variable by name from an async context.
+
+    ```
+        from prefect import variables
+
+        @flow
+        async def my_flow():
+            var = await variables.aget("my_var")
+    ```
+    """
+    variable = await _get_variable_by_name(name)
+    return variable.value if variable else default
+
+
+async def _get_variable_by_name(name):
+    async with get_client() as client:
```

Comment:
Should probably use `inject_client` so it receives optimizations to client retrieval